### ISSUE TITLE
Muse - museMonitor format

### ISSUE THREAD

**Issue by francescacoo** [curiositry/EEGrunt#17]
Hi,
it might be too much to ask, but I try anyway!

I have a muse 2016 and unfortunately I am having problems running muse-lsl for bluetooth issues on my machine :(

So I am using an app called MuseMonitor and I can record in .muse format or .csv.
When I try to run the analyze_data I get "IndexError: too many indices for array"

I realize that the problem is the format for sure different..
I am beginner with python, so I will try to check if I can fix the problem myself, but if you have any suggestion it will for sure help!

the column names I get from the .csv and a first line:

`TimeStamp,Delta_TP9,Delta_AF7,Delta_AF8,Delta_TP10,Theta_TP9,Theta_AF7,Theta_AF8,Theta_TP10,Alpha_TP9,Alpha_AF7,Alpha_AF8,Alpha_TP10,Beta_TP9,Beta_AF7,Beta_AF8,Beta_TP10,Gamma_TP9,Gamma_AF7,Gamma_AF8,Gamma_TP10,RAW_TP9,RAW_AF7,RAW_AF8,RAW_TP10,AUX_RIGHT,Accelerometer_X,Accelerometer_Y,Accelerometer_Z,Gyro_X,Gyro_Y,Gyro_Z,HeadBandOn,HSI_TP9,HSI_AF7,HSI_AF8,HSI_TP10,Battery,Elements
2017-11-20 22:41:01.738,0.9025755,0.6792804,0.0,1.0303508,0.6843497,0.3073409,0.0,0.7147551,1.1894307,0.41819534,0.0,0.9815956,0.7329064,0.1080209,0.0,0.5908704,0.3168565,-0.15387148,0.0,0.16846296,856.2271,946.4835,747.4359,864.68866,954.5421,0.429931640625,0.12445068359375,0.90252685546875,-1.51031494140625,6.287994384765625,2.51220703125,1,1.0,1.0,2.0,1.0,97.94
`
If you need I could send you a recording!
(if I solve it in the meanwhile I will let you know!)

Thank you!

**Comment by curiositry** [comment]
@francescacoo Pardon the slow reply. At the moment EEGrunt only supports plain Muse data, and support for data recorded with museMonitor isn’t currently on the roadmap. However, it may not be too difficult to write a Python script to convert museMonitor recordings into something EEGrunt can digest.

If you’d like to send a recording, you can either uploaded it here or send it to info {AT} autodidacts {DOT} io . 

@Photosynthesis and I haven’t had much time to put into EEGrunt lately, so I can’t guarantee anything, but but we could at least take a look :)

**Comment by akanshmurthy** [comment]
Hi, @curiositry . I have a similar question as I get the same error even with a different format. What do you mean by plain Muse data? Could you provide a sample of that somewhere in the documentation (i.e. what types/numbers of columns are you expecting?)?

**Comment by curiositry** [comment]
@akanshmurthy Here’s a sample: https://neurobb.com/t/eegrunt-support-for-muse-headset-data/96/6?u=curiositry

**Comment by akanshmurthy** [comment]
@curiositry : thanks for that. That file is pretty hard to parse though. There are no labels for the columns. Are the columns timestamp, electrode, and voltage value (but there are multiple columns for the voltage and I even see fft, which I don't think is part of the raw Muse data...)? 


I have a few different formats from a single recording: 

- the raw Muse format, which involves columns for each electrode and other measures such as accelerometer and gyro data
- a file with just 2 columns (1 for timestamp and the other for voltage which includes voltages for multiple electrodes)
- a file with 5 columns (1 for timestamp and 4 that includes one for each electrode)

What format are you expecting?

**Comment by akanshmurthy** [comment]
I think I figured this issue out. @francescacoo : use this library to record your data (https://github.com/alexandrebarachant/muse-lsl). Or, alternatively, remove all the extra columns from your Muse Monitor data until you only have the 4 columns with the raw data from the 4 electrodes. Essentially, you want to force the pathway in load_data of EEGrunt.py to use the muse-lsl source.

**Comment by E3V3A** [comment]
Could someone provide some raw muse sample data? (According to @akanshmurthy  description?)
